hudi-hadoop-mr/src/main/java/org/apache/hudi/hadoop/utils/HoodieRealtimeRecordReaderUtils.java

```diff
@@ -189,7 +190,13 @@ public static Writable avroToArrayWritable(Object value, Schema schema) {
         Writable[] recordValues = new Writable[schema.getFields().size()];
         int recordValueIndex = 0;
         for (Schema.Field field : schema.getFields()) {
-          recordValues[recordValueIndex++] = avroToArrayWritable(record.get(field.name()), field.schema());
+          Object fieldValue = null;
+          try {
+            fieldValue = record.get(field.name());
+          } catch (AvroRuntimeException e) {
+            LOG.debug("Field:" + field.name() + "not found in Schema:" + schema.toString());
```

Comment:
If the field is not found, should that fail the conversion instead of filling `null`?

Reply:
I believe the field not being found in this case is normal based on the use case mentioned in the detailed pr description. 

cc @zhedoubushishi

Reply:
@yihua The way things are currently implemented, is that this function is supposed to return a record with complete schema. We cannot fail if the field is not found, as it is required for both bootstrap and schema evolution scenarios. In case of bootstrap, the metadata fields may not be found in the data file and need to be filled with nulls. Similarly with schema evolution, we can hit a scenario like this and historically what we do is return nulls for the new columns if the old record does not have them.

Reply:
It makes sense to me now.  If the existing behavior is to return null, then this is OK.  @rahil-c could you create a ticket for revisiting the performance?